templates/test/ci/cluster-template-prow.yaml

```diff
@@ -5,6 +5,7 @@ metadata:
     cni-windows: ${CLUSTER_NAME}-calico
     containerd-logger: enabled
     csi-proxy: enabled
+    windows-service-proxy: kube-proxy
```

Comment:
The new kpng image is really new and we don't have sufficient test coverage to ensure that it will pass all the upstream tests.  This is one step in that direction.

Reply:
got it, in that case do we still need the additional test template? since the prow test template has it? or maybe it's the other way around if we still want to test both, should we keep the default prow template untouched?

Reply:
@CecileRobertMichon 
Right now only the new `-windows-kpng.yaml` template will install KPNG. All of the other templates will install kube-proxy.

I staged this PR into 2 commit. 
The first commit decouples CNI installation and kube-proxy installation (which is why `windows-service-proxy: kube-proxy` is getting added to all of the templates) because previous we would install in-tree kubeproxy on all templates that set `cni-windows`)
The second commit adds a new add to install kpng IF `windows-service-proxy: kpng` label is set for the cluster (which should only happen on the new `-windows-kpng.yaml` template.

Reply:
I do think we want to be able to test both for a while.

Reply:
thanks that clarifies